mkosi/config.py

```diff
@@ -445,6 +445,23 @@ def native(cls) -> "Architecture":
         return cls.from_uname(platform.machine())
 
 
+class CapitalisedKeyDict(dict[str, Any]):
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__()
+        for k, v in dict(*args).items():
+            self[k] = v
+        for k, v in kwargs.items():
+            self[k] = v
+
+    def __setitem__(self, key: str, value: Any) -> None:
+        def key_transformer(k: str) -> str:
+            if (s := SETTINGS_LOOKUP_BY_DEST.get(k)) is not None:
+                return s.name
+            return "".join(p.capitalize() for p in k.split("_"))
+
+        super().__setitem__(key_transformer(key), value)
```

Comment:
I wouldn't bother with subclassing dict. A factory function that simply initializes a regular dict with the key transformer is fine.